src/Impersonator.php

```diff
@@ -427,4 +444,24 @@ protected function dispatchEventAfterResponse(object $event): void
             $result->afterResponse();
         }
     }
+
+    /**
+     * Get the decrypted recaller cookie for the request.
+     */
+    protected function recaller(Guard|StatefulGuard $guard): ?Recaller
+    {
+        if (
+            ! is_callable([$guard, 'getRequest'])
+            || ! is_callable([$guard, 'getRecallerName'])
+            || is_null($guard->getRequest())
+        ) {
+            return null;
+        }
```

Comment:
```suggestion
    protected function recaller(SessionGuard $guard): ?Recaller
    {
        if (is_null($guard->getRequest())) {
            return null;
        }
```

As the remember token only works with `SessionGuard`, I think we should type–hint the `$guard` parameter accordingly. We also don’t need to check both callables.